### PR TITLE
[RAPTOR-5931] R transform template

### PR DESCRIPTION
**Pull request by jmbannon**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Add an R transform template. In addition, scrub the `y` parameter from transform functions and update documentation.

### CODE REVIEW

**Comment by jmbannon** [comment]
jarvis test functional_inference_model_templates functional_training_model_templates

**Review by 4dahalibut — APPROVED** [review]

File: task_templates/transforms/r_transform_recipe/custom.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#   transformer: Any
+#       Trained object, extracted by DataRobot from the artifact created in fit().
+#       In this example, contains the trained recipe extracted from r_transform_recipe.rds.
+#
+#   Returns
+#   -------
+#   data.frame
+#       Returns the transformed dataframe
+#   "
+#   bake(transformer, X)
```
Comment (4dahalibut) [inline task_templates/transforms/r_transform_recipe/custom.R:91]: howcome this is commented out?

Comment (jmbannon) [reply task_templates/transforms/r_transform_recipe/custom.R:91]: because it bakes recipes by default, so this isn't actually needed

Comment (jmbannon) [reply task_templates/transforms/r_transform_recipe/custom.R:91]: if the artifact is a recipe

**Review by meltzg — APPROVED** [review]

**Pull request merged**